packages/medusa/src/api/routes/store/variants/get-variant.ts

```diff
@@ -82,11 +80,8 @@ export default async (req, res) => {
   })
 
   let sales_channel_id = validated.sales_channel_id
-  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
-  if (featureFlagRouter.isFeatureEnabled(PublishableAPIKeysFeatureFlag.key)) {
-    if (req.publishableApiKeyScopes?.sales_channel_id.length === 1) {
-      sales_channel_id = req.publishableApiKeyScopes.sales_channel_id[0]
-    }
+  if (req.publishableApiKeyScopes?.sales_channel_id.length === 1) {
```

Comment:
question: We are swallowing the error if there is more than one item. Should we throw like the other places?

Reply:
currently, we are only throwing when a cart is created if there are multiple sales channels, I guess it depends on the use case, but since this is related to the inventory feature I'm not sure